src/shared/Core/Trace2.cs

```diff
@@ -481,3 +504,16 @@ private void WriteMessage(Trace2Message message)
         }
     }
 }
+
+public static class ThreadHelpers
+{
+    public static string BuildThreadName()
```

Comment:
We are passing `ThreadHelpers.BuildThreadName()` in almost all of these places.. should we just not always use the `BuildThreadName()` logic inside each `WriteX` method?

To capture all possible thread names, including `main` what about this?

```csharp
// If the thread is named, is it!
if (!string.IsNullOrEmpty(Thread.CurrentThread.Name))
{
    return Thread.CurrentThread.Name;
}

// Otherwise if this is the entry thread, call it "main"
if (Thread.CurrentThread.ManagedThreadId == 0)
{
    return "main";
}

// If this is a thread pool thread, name it as such
if (Thread.CurrentThread.IsThreadPoolThread)
{
    return $"thread_pool_{Environment.CurrentManagedThreadId}";
}

// We don't know what this thread is!
return string.Empty;
```

Reply:
Nice! Thanks for the much improved suggestion.